testflight/admin_priviliges_test.go

```diff
@@ -0,0 +1,173 @@
+package testflight_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("Admin priviliges", func() {
+	var tmpDir, ogHome, oldTarget, pipelineName, pipelineConfig, resourceName, resourceTypeName, jobName, taskName string
+	priviligedAdminTarget := testflightFlyTarget + "-padmin"
+	newTeamName := "priviliged-admin-test-team"
+
+	BeforeEach(func() {
+		var err error
+		tmpDir, err = ioutil.TempDir("", "fly-test")
+		Expect(err).ToNot(HaveOccurred())
+
+		ogHome = os.Getenv("HOME")
+		os.Setenv("HOME", tmpDir)
+
+		Eventually(func() *gexec.Session {
+			login := spawnFlyLogin(adminFlyTarget)
+			<-login.Exited
+			return login
+		}, 2*time.Minute, time.Second).Should(gexec.Exit(0))
+
+		pipelineName = randomPipelineName()
+		resourceName = "time-test-resource"
+		resourceTypeName = "my-time"
+		jobName = "admin-sample-job"
+		taskName = "simple-task"
+
+		pipelineConfig = filepath.Join(tmpDir, "pipeline.yml")
+
+		err = ioutil.WriteFile(pipelineConfig,
+			[]byte(`---
+resource_types:
+- name: `+resourceTypeName+`
+  type: registry-image
+  source: {repository: concourse/time-resource}
+
+resources:
+- name: `+resourceName+`
+  type: my-time
+  source: {interval: 1h}
+
+jobs:
+  - name: `+jobName+`
+    public: true
+    plan:
+      - get: `+resourceName+`
+      - task: `+taskName+`
+        config:
+          platform: linux
+          image_resource:
+            type: registry-image
+            source: { repository: busybox }
+          run:
+            path: /bin/sh
+            args:
+            - -c
+            - |
+              until test -f /tmp/stop-waiting; do
+                echo 'waiting for /tmp/stop-waiting to exist'
+                sleep 1
+              done
+              sleep 100
+              echo done
+              `), 0644)
+		Expect(err).NotTo(HaveOccurred())
+
+		oldTarget = flyTarget
+		flyTarget = adminFlyTarget
```

Comment:
mutating this unprotected global variable in a multithreaded context is very dangerous, and I think it is causing the failures we see on CI, where this test sets the variable in one thread, interfering with the other threads' ability to log in. I see this as a bit of a testing disaster that is nobody's fault 😢 ... I suspect it is safer to put something like this in the `SynchronizedBeforeSuite` and have this target just stick around for the duration of the whole testflight suite? there is also a `withFlyTarget` helper, which is also not thread-safe...